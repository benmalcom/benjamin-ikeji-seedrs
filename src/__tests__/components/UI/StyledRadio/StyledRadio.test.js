import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import StyledRadio from 'components/UI/StyledRadio';

describe('StyledRadio', () => {
  const getComponent = props => render(<StyledRadio {...props} />);

  it('renders without crash', async () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders with expected functionality', async () => {
    const onClick = jest.fn();
    const props = {
      onClick,
      "data-testid": "check",
      checked: false,
      value: 1
    };
    const { getByTestId } = getComponent(props);
    const styledRadio = getByTestId('check');
    fireEvent.click(styledRadio, { target: { checked: true }});
    expect(styledRadio).toHaveProperty('checked', true);
  });

});