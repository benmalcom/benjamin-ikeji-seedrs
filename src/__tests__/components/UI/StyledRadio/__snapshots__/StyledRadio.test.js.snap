// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledRadio renders without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root makeStyles-root-1"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-7"
            type="radio"
            value=""
          />
          <span
            class="makeStyles-icon-2"
          />
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiRadio-root makeStyles-root-1"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-7"
          type="radio"
          value=""
        />
        <span
          class="makeStyles-icon-2"
        />
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
